js/script.js

```diff
@@ -0,0 +1,104 @@
+var counter = 0;
+var intervalId;
+
+$(document).ready(function() {
+
+	$('#reddit-search').on('click', function(e) {
+	    e.preventDefault();
+	    search($('#input').val());
+	    console.log('clicked'+$('#input').val())
+	    $('#input').val('');
+  	})
+
+  		$('#stop-search').on('click', function(e) {
+			    e.preventDefault();
+			    $('#images').html('');
+
+			    clearInterval(intervalId);
+
+  			})
+
+	function search(input) {
+		$.get('http://www.reddit.com/search.json', {q:input},"+nsfw:no").done(function(data) {
+	   		console.log(data);
+
+	   		var children = data.data.children;
+	   		console.log(children);
+
+	   		var hasUrl = function(child){
+	   			if(child.data.url){
+	   				return true;
+	   			} else {
+	   				return false;
+	   			}
+	   		};
+
+	   		var filteredChildren = children.filter(hasUrl);
+	   		
+	   		var imageArray = filteredChildren.map(function(img) {
+				return img.data.url;
+		   	});
+
+	   		intervalId =setInterval(function() {
+		     $('#images').html('').append('<img src="'+imageArray[counter]+'">');
+		      
+		     
+
+		      counter++;
+		      // if (counter >= doughnuts.length) {
+		      //   counter = 0;
+		      // }
+		      counter = counter % imageArray.length;
+
+		   	}, 2000);
+
+		   
+		});		 // console.log(imageArray);
+	};
+
+	
+	// //filter the children, making sure each child has an image url (hint data.url)
```

Comment:
Try not to leave commented-out code in your files. If you need to look at a pervious version of your code, that's what source control is for. Look at your previpous commits. If it's not in a previous commit, then commit more!